addons/mail/models/mail_notification.py

```diff
@@ -16,6 +16,7 @@ class MailNotification(models.Model):
     _description = 'Message Notifications'
 
     # origin
+    author_id = fields.Many2one('res.partner', "Author", related='mail_message_id.author_id', store=True, ondelete='set null')
```

Comment:
Why a related ? I don't think we need an onchange-like mechanism. As notifications are always created manually I would update their creation (which is what you already did). Anyway we never update a message author, except in the infamous "post message post hook" that tries to do it after partner creation. We could make a small addition in that method instead of relying on costly related fields. What do you think ?

Reply:
Having a related is stronger in terms of data consistency, it's hard to remember to fill the value manually when creating new code (and even harder to find all occurrences without a doubt in existing code, which doesn't even touch on the problem of custom code potentially creating notifications either on internal or from customers).

If the only reason to not have related are cost of related, shouldn't related be optimized to not be so costly instead?
And only if there is a measurable issue that comes with it?

It's unfortunate that it adds a query on `unlink`, but as I posed on the original message it should be fixable.
And I don't think the cost in terms of update matters much, as we probably never update message on notification, and we almost never update author on message either.

Reply:
Related are computed fields and have high costs we have to take into account. You did half the job adding it in notification creation, let us finish it. This model is technical, I don't want it to have computed fields involving write, check of triggers, ... when we can avoid them.

About optimization of related fields: sure but it will not be done. We can set a related once those are optimized. Better fix locally than wait for an hypothetical improvement that is not planned.

Reply:
I'll remove the related because I don't want to block the PR for it, but I don't like that we are making the business code more complex to circumvent issues with the ORM. But fair enough to avoid using it until/if `related` becomes more desirable to use in this case.